python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_chat_completion.py

```diff
@@ -218,6 +209,102 @@ def _create_chat_message_content(
             metadata=metadata,
         )
 
+    # endregion
+
+    # region Streaming
+    async def get_streaming_chat_message_contents(
+        self,
+        chat_history: ChatHistory,
+        settings: AzureAIInferenceChatPromptExecutionSettings,
+        **kwargs: Any,
+    ) -> AsyncGenerator[list[StreamingChatMessageContent], Any]:
+        """Get streaming chat message contents from the Azure AI Inference service.
+
+        Args:
+            chat_history: A list of chats in a chat_history object.
+            settings: Settings for the request.
+            kwargs: Optional arguments.
+
+        Returns:
+            A list of chat message contents.
+        """
+        if (
+            settings.function_choice_behavior is None
+            or not settings.function_choice_behavior.auto_invoke_kernel_functions
+        ):
+            # No auto invoke is required.
+            async_generator = self._send_chat_streaming_request(chat_history, settings)
+        else:
+            # Auto invoke is required.
+            async_generator = self._get_streaming_chat_message_contents_auto_invoke(chat_history, settings, **kwargs)
+
+        async for messages in async_generator:
+            yield messages
+
+    async def _get_streaming_chat_message_contents_auto_invoke(
```

Comment:
Is `auto_invoke` the correct nomenclature here? Shouldn't it be `tool_call` or `function_call`? Just because auto invoke is False, we can still handle `FunctionCallContent`, which means we need to process functions/tools.

Reply:
The default behavior will handle FunctionCallContent. The helper function will execute the function, thus auto-invoke.